src/java.base/share/classes/com/sun/crypto/provider/GaloisCounterMode.java

```diff
@@ -726,6 +767,16 @@ int decryptFinal(byte[] in, int inOfs, int len,
     int decryptFinal(ByteBuffer src, ByteBuffer dst)
         throws IllegalBlockSizeException, AEADBadTagException,
         ShortBufferException {
+
+        // Check that the dst buffer is large enough for the plaintext.  This
+        // is done here as gctr requires extra bytes to complete a blockSize
+        // The unnecessary extra bytes can cause buffers with enough size for
+        // the data to fail with a short buffer exception
+        if ((src.remaining() + ((ibuffer != null) ? ibuffer.size() : 0) -
+            tagLenBytes) > dst.remaining()) {
+            throw new RuntimeException("output buffer too small");
```

Comment:
Shouldn't this be ShortBufferException instead of RuntimeException?

Reply:
I thought so too, but that isn't what GCTR returns.  All the GCTR checks are RuntimeExceptions.  This check was original inside of GCTR, but I had to bring it out because of the ibuffer lengths.  I don't mind changing, it's just a strange inconsistency.

Reply:
GaloisCounterMode is the caller class of GCTR and this API is meant to throw ShortBufferException I think (based on the check at lines below at 820-822)...

Reply:
Yes.. this check is not necessary given the ShortBufferException was already checked.  I am removing this as part of the next update